source/firmware/arch/timing-manager.md

```diff
@@ -0,0 +1,170 @@
+
+
+### Timing Diagram
+
+The following diagram shows the behaviour of these events: triggers, sensors reporting as done, and generation of scheduler interrupts:
+
+```{figure} images/timing.png
+:alt: Timing Manager timing diagram
+:figwidth: 100%
+:width: 600em
+:align: left
+```
+
+The diagram shows how the Timing Manager receives the PWM Carrier high and low events. The diagram shows the operation for the Timing Manager configured to trigger every three carrier valleys (trigger on `carrier_low` only, with a 3:1 ratio).
+
+When the first `carrier_low` occurs, the timing manager flushes the PWM duty ratios and triggers the three enabled sensor interfaces to begin sampling. Sensor 1 reports that it is done acquiring new data practically immediately, with Sensor 3 reporting done shortly thereafter. 
+
+After a while, Sensor 2 reports done as well. Since all three sensors have reported done (`all_done`), the timing manager generates the scheduler interrupt from the FPGA to the ARM processor, and the scheduler code takes control of the processor and begins scheduling control task code to run.
+
+The control code will run, update the PWM duty ratios, and eventually yield the processor back to the scheduler. Ideally, all tasks that need to be scheduled to run in this timeslice will complete with some time left before the third (the 3:1 ratio, in this example) `carrier_low` event, where the cycle will begin again. If the tasks are not all able to complete, the ratio may have to be increased.
+
+
+## Triggering Modes
+
+The Timing Manager can be configured to trigger sensors in one of two modes:
+- Automatic - triggering on every valid PWM carrier event (as defined by the PWM peak/valley configuration and the User Event Ratio) automatically **(default)**
+- Manual - a single trigger mode useful for debugging, where the user can send a command via the command interface to queue a single trigger, which will be sent out by the Timing Manager at only the next qualifying event
+
+This configuration can be done via the C driver interface (given below).
+
+
+## Scheduler ISR Modes
+
+The Timing Manager can be configures to generate a scheduler interrupt in one of two modes:
+- "Legacy" Mode: The scheduler interrupt is generated only based on the PWM configuation and user ratio, and does not wait for sensor interfaces to report `done`. To maintain backwards-compatibility, this is the **default** setting. 
+- "New" Mode: The scheduler interrupt is only generated when all sensors enabled via the Timing Manager report `done`. If no sensors are enabled, the scheduler interrupt is generated in the way as the Legacy Mode.
+
+This configuration must be set before code compilation and flashing using the `USER_CONFIG_ISR_SOURCE` definition in `usr/user_config.h`.
```

Comment:
@Known4225 I wrote this a while back and would love if you could fix a few things for me.

configures -> configured

"New" Mode -> "Post-Sensor" Mode (please ctrl-F through all the new pages in this PR to make sure the name of the new, Post-Sensor Mode is consistent everywhere. Sigh... we will also likely need to update the code itself (`user_config.h`) in the AMDC-Firmware repo. Could you open an issue in that repo to serve as a reminder to make that rename change?